### PR TITLE
[SPARK-37456][SQL] CREATE NAMESPACE should qualify location for v2 command

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

v2 `CREATE NAMESPACE` command currently doesn't qualify the location property whereas v1 command qualifies it. This PR proposes to make the behavior consistent by qualifying the location property for v2 command.

### Why are the changes needed?

To make the behavior consistent b/w v1 and v2 command, but not qualifying the location property seems like a bug since the default filesystem can change.

### Does this PR introduce _any_ user-facing change?

Yes, now the location property will be qualified before it is passed to v2 catalog.

### How was this patch tested?

Added a new test.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 1]
```diff
@@ -375,6 +375,11 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
       val path = new Path(tmpDir.getCanonicalPath).toUri
       databaseNames.foreach { dbName =>
         try {
+          val e = intercept[IllegalArgumentException] {
+            sql(s"CREATE DATABASE $dbName Location ''")
+          }
+          assert(e.getMessage.contains("Can not create a Path from an empty string"))
```
Comment (imback82) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala:381]: This is just adding a test coverage for v1 command.

**Comment by imback82** [comment]
cc @cloud-fan

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
retest please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**